strr-api/src/strr_api/resources/registrations.py

```diff
@@ -312,6 +314,58 @@ def update_registration_status(registration_id):
         return error_response(message=ErrorMessage.PROCESSING_ERROR.value, http_status=HTTPStatus.INTERNAL_SERVER_ERROR)
 
 
+@bp.route("/<registration_id>/todos", methods=("GET",))
+@swag_from({"security": [{"Bearer": []}]})
+@cross_origin(origin="*")
+@jwt.requires_auth
+def get_todos(registration_id):
+    """
+    Get todos for a registration.
+    ---
+    parameters:
+      - in: path
+        name: registration_id
+        type: integer
+        required: true
+        description: Registration Id
+    responses:
+      200:
+        description: Success
+      401:
+        description: Unauthorized to retrieve the todos for the registration.
+      500:
+        description: Unexpected error during processing.
+    """
+    try:
+        account_id = request.headers.get("Account-Id")
+        user = User.get_or_create_user_by_jwt(g.jwt_oidc_token_info)
+        if not user:
+            raise AuthException()
+        registration = RegistrationService.get_registration(account_id, registration_id)
+        if not registration:
+            raise AuthException()
+
+        todos = []
+
+        if registration.status in [RegistrationStatus.ACTIVE]:
```

Comment:
I think there should be another status here. Expired registrations will also have a Renewal To Do (to allow Host to renew within the 3 year after expiration).